packages/taco/src/config-validator.ts

```diff
@@ -0,0 +1,279 @@
+   * @returns {Promise<boolean>} True if provider is valid for domain
+   */
+  static async isValidProvider(
+    domain: DomainName,
+    provider: PublicClient | ethers.providers.Provider,
+  ): Promise<boolean> {
+    let chainId: number;
+
+    if (!provider || typeof provider !== 'object') {
+      // Invalid provider
+      return false;
+    }
+
+    // Try to detect provider type and get chain ID safely
+    try {
+      if (isViemClient(provider)) {
+        chainId = await (provider as PublicClient).getChainId();
+      } else {
+        const network = await (
+          provider as ethers.providers.Provider
+        ).getNetwork();
+        chainId = network.chainId;
+      }
+    } catch (error) {
+      // Error getting chain ID
+      return false;
+    }
+
+    // Check if the provider's chain ID matches the domain's expected chain ID
+    return (
+      Object.values(DOMAINS).find(
+        (domainInfo) =>
+          domainInfo.domain === domain && domainInfo.chainId === chainId,
+      ) !== undefined
+    );
+  }
+
+  /**
+   * Fast validation (everything except provider network checks)
+   *
+   * Performs synchronous validation of configuration including:
+   * - Domain name validation
+   * - Ritual ID validation to ensure it is a positive integer
+   * - Provider/signer presence validation
+   * - Chain compatibility check (if chain info is available synchronously)
+   *
+   * @param {TacoClientConfig} config - Configuration to validate
+   * @returns {ValidationResult} Validation result with isValid boolean and errors array
+   */
+  static validateFast(config: TacoClientConfig): ValidationResult {
+    const errors: string[] = [];
+
+    // Validate domain
+    if (!config.domain) {
+      errors.push('The property `domain` is required');
+    } else if (!this.isValidDomain(config.domain)) {
+      errors.push(
+        `Invalid domain name: ${config.domain}. Supported domains: ${this.getSupportedDomains().join(', ')}`,
+      );
+    }
+
+    // Validate ritual ID
+    if (!config.ritualId) {
+      errors.push('The property `ritualId` is required');
+    } else if (!this.isValidRitualId(config.ritualId)) {
+      errors.push(
+        `Invalid ritual ID: ${config.ritualId} for domain ${config.domain}`,
+      );
+    }
+
+    // Validate blockchain client configuration
+    if ('viemClient' in config) {
+      // Viem configuration
+      if (!config.viemClient) {
+        errors.push('viemClient is required for viem configuration');
+      }
+      if (!config.viemSignerAccount) {
+        errors.push('viemSignerAccount is required for viem configuration');
+      }
+    } else if ('ethersProvider' in config) {
+      // Ethers configuration
+      if (!config.ethersProvider) {
+        errors.push('ethersProvider is required for ethers configuration');
+      }
+      if (!config.ethersSigner) {
+        errors.push('ethersSigner is required for ethers configuration');
+      }
```

Comment:
Can some of this be made simpler by utilizing `isViemConfig` and `isEthersConfig` from client-config.ts?

Reply:
Here it is something else. In the class constructor, I used named parameters, in order to enforce an explicit pass of either ethers or viem objects with no ambiguous types.

Named parameters are popular in JavaScript/TypeScript for functions that take many parameters. Because it is handy, there is an eslint rule for that: https://eslint.org/docs/latest/rules/max-params. The rule is not enabled by default. But if enabled, the default value for the max parameters is 3.